geomesa-datastore-bundle/geomesa-datastore-processors/src/main/scala/org/geomesa/nifi/datastore/processor/mixins/FeatureTypeProcessor.scala

```diff
@@ -28,55 +28,55 @@ import org.opengis.feature.simple.SimpleFeatureType
     new ReadsAttribute(attribute = "geomesa.sft.spec", description = "GeoMesa SimpleFeatureType specification")
   )
 )
-trait FeatureTypeProcessor extends BaseProcessor {
+trait FeatureTypeProcessor extends UserDataProcessor {
 
   import FeatureTypeProcessor.Attributes.{SftNameAttribute, SftSpecAttribute}
   import FeatureTypeProcessor.Properties.{FeatureNameOverride, SftNameKey, SftSpec}
 
   private var sftName: PropertyDescriptor = _
 
-  private val sftCache = Caffeine.newBuilder().build(
+  private val sftCache = Caffeine.newBuilder().build[SftArgs, Either[Throwable, SimpleFeatureType]](
     new CacheLoader[SftArgs, Either[Throwable, SimpleFeatureType]]() {
-      override def load(key: SftArgs): Either[Throwable, SimpleFeatureType] =
-        SftArgResolver.getArg(key).right.map(decorate)
+      override def load(key: SftArgs): Either[Throwable, SimpleFeatureType] = SftArgResolver.getArg(key)
     }
   )
 
   override protected def getPrimaryProperties: Seq[PropertyDescriptor] = {
     sftName = FeatureTypeProcessor.Properties.sftName(SimpleFeatureTypeLoader.listTypeNames)
-    super.getPrimaryProperties ++ Seq(sftName, SftSpec, FeatureNameOverride)
+    Seq(sftName, SftSpec, FeatureNameOverride) ++ super.getPrimaryProperties
```

Comment:
Does this change matter?  (I imagine it does; just asking to understand why.)

Reply:
it just changes the order in the UI... this puts the 'feature type user data' property after the feature type ones